lib/extensions/method.rb

```diff
@@ -145,11 +157,25 @@ def invalid_args?
         pair.first == :req
       end
 
-      too_many = args.size > unnamed_params.size
+      unlimited = unnamed_params.any? do |pair|
+        pair.first == :rest
+      end
+
+      too_many = !unlimited && args.size > unnamed_params.size
```

Comment:
Check the source method to see if it takes unbounded arguments (*args), if it does then any number of arguments are allowed here as long as the required arguments have been supplied.